docker/docker_image_dest.go

```diff
@@ -596,6 +637,140 @@ func (d *dockerImageDestination) putOneSignature(url *url.URL, signature []byte)
 	}
 }
 
+func (d *dockerImageDestination) putSignaturesToCosignAttachments(ctx context.Context, signatures []signature.Cosign, manifestDigest digest.Digest) error {
+	if !d.c.useCosignAttachments {
+		return errors.New("writing Cosign attachments is disabled by configuration")
+	}
+
+	ociManifest, err := d.c.getCosignAttachmentManifest(ctx, d.ref, manifestDigest)
+	if err != nil {
+		return nil
+	}
+	var ociConfig imgspecv1.Image // Most fields empty by default
```

Comment:
TO DO: Only fetch the manifest if we actually have a new signature to add (and if we continue to update the DiffID list).